internal/featuredetection/feature_detection_test.go

```diff
@@ -348,3 +350,83 @@ func TestRepositoryFeatures(t *testing.T) {
 		})
 	}
 }
+
+func TestProjectV1SupportEnvOverride(t *testing.T) {
+	tt := []struct {
+		name            string
+		envVal          string
+		expectedSupport gh.ProjectsV1Support
+		expectedErr     bool
+	}{
+		{
+			name:            "PROJECTS_V1_SUPPORTED=true",
+			envVal:          "true",
+			expectedSupport: gh.ProjectsV1Supported,
+		},
+		{
+			name:            "PROJECTS_V1_SUPPORTED=false",
+			envVal:          "false",
+			expectedSupport: gh.ProjectsV1Unsupported,
+		},
+		{
+			name:        "PROJECTS_V1_SUPPORTED=nonboolean",
+			envVal:      "nonboolean",
+			expectedErr: true,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Setenv("PROJECTS_V1_SUPPORTED", tc.envVal)
+
+			detector := detector{}
+			support, err := detector.ProjectsV1()
+
+			if tc.expectedErr {
+				require.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedSupport, support)
+		})
+	}
+}
+
+func TestProjectV1APINotSupported(t *testing.T) {
+	// Given the API does not include projects in the returned fields
+	reg := &httpmock.Registry{}
+	httpClient := &http.Client{}
+	httpmock.ReplaceTripper(httpClient, reg)
+	reg.Register(
+		httpmock.GraphQL(`query ProjectsV1FeatureDetection\b`),
+		httpmock.StringResponse(heredoc.Doc(`{ "data": { "Repository": { "fields": [] } } }`)),
+	)
+
+	// When we detect ProjectV1 support
```

Comment:
What I think you're trying to say is "this is where we are checking the api if there is ProjectV1 support". What this reads to me as is "we have detected that ProjectV1 is supported".

```suggestion
	// When we call the api to detect for ProjectV1 support
```

Reply:
Thanks, addressed.